agent/api/statechange.go

```diff
@@ -105,6 +105,9 @@ type AttachmentStateChange struct {
 // returns error if the state change doesn't need to be sent to the ECS backend.
 func NewTaskStateChangeEvent(task *apitask.Task, reason string) (TaskStateChange, error) {
 	var event TaskStateChange
+	if task.IsInternal {
+		return event, errors.Errorf("skip creating task stage change event for internal task %v", task.Arn)
```

Comment:
Since it is expected, should we return error for this ? Will it be better to add a debug level log and returning nil error?

Reply:
Callers ([here](https://github.com/aws/amazon-ecs-agent/blob/41e8e6a78391f65b17070ffe086876e63387d5bb/agent/engine/task_manager.go#L617) and [here](https://github.com/aws/amazon-ecs-agent/blob/fe0ce622b35294c773509831a31809910667bdb2/agent/engine/docker_task_engine.go#L786)) currently use error value for determining whether an event should be sent. This pattern applies to other types of event publishing as well. It might be better to have a separate `shouldCreateXXXEvent` method to be more clear, but I don't have strong objection with our current approach either (where we treat the "should not" situation as error).

Reply:
makes sense.